spec/services/exports/export_donations_csv_service_spec.rb

```diff
@@ -2,23 +2,63 @@
   describe '#generate_csv_data' do
     subject { described_class.new(donation_ids: donation_ids).generate_csv_data }
     let(:donation_ids) { donations.map(&:id) }
+    let(:duplicate_item) do
+      FactoryBot.create(
+        :item, name: Faker::Appliance.equipment
+      )
+    end
+    let(:items_lists) do
+      [
+        [
+          [duplicate_item, 5],
+          [
+            FactoryBot.create(
+              :item, name: Faker::Appliance.equipment
+            ),
+            7
+          ],
+          [duplicate_item, 3]
+        ],
+        *(Array.new(3) do |i|
+          [[FactoryBot.create(
+            :item, name: Faker::Appliance.equipment
+          ), i + 1]]
+        end)
+      ]
+    end
+
+    let(:item_names) { items_lists.flatten(1).map(&:first).map(&:name).sort.uniq }
     let(:donations) do
       start_time = Time.current
-      Array.new(3) do |i|
-        create(
+
+      items_lists.each_with_index.map do |items, i|
+        # We're gonna mutate for dubious profit.
+        items = items.dup
+        # ¡MUTATE! 🎉
+        (item, quantity) = items.shift
+
+        donation = create(
           :donation,
           :with_items,
           donation_site: create(
             :donation_site, name: "Space Needle #{i}",
           ),
-          item: FactoryBot.create(
-            :item, name: Faker::Appliance.equipment
-          ),
+          item: item,
+          item_quantity: quantity,
           issued_at: start_time + i.days,
           comment: "This is the #{i}-th donation in the test."
         )
+
+        items.each do |(item, quantity)|
+          donation.line_items << create(
+            :line_item, quantity: quantity, item: item, itemizable: donation
+          )
```

Comment:
FYI you don't need lines 53 and 55 since you're passing the donation into the `create` call.